src/imgtools/cli/autopipeline.py

```
@@ -0,0 +1,267 @@
+
+def parse_spacing(ctx, param, value): # type: ignore
+    """Parse spacing as a tuple of floats."""
+    if not value:
+        return (0.0, 0.0, 0.0)
+    try:
+        parts = value.split(",")
+        if len(parts) != 3:
+            raise click.BadParameter("Spacing must be three comma-separated values")
+        return tuple(float(p) for p in parts)
+    except ValueError:
+        raise click.BadParameter("Spacing values must be valid floats")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)